### PR TITLE
[PT FE] Added support for aten::logit and aten::special_logit

### PR DESCRIPTION
**Pull request by code-dev05**
### Details:
 - *added aten::logit*
 - *added aten::special_logit*

### Tickets:
 - *[28744](https://github.com/openvinotoolkit/openvino/issues/28744#issuecomment-2717666789)*

### CODE REVIEW

File: src/frontends/pytorch/src/op/logit.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    auto one = context.mark_node(v0::Constant::create(element::f32, Shape{}, {1}));
+    one = context.mark_node(std::make_shared<v1::ConvertLike>(one, x));
+    auto one_minus_x = context.mark_node(std::make_shared<v1::Subtract>(one, x));
+    auto division = context.mark_node(std::make_shared<v1::Divide>(x, one_minus_x));
+    return {context.mark_node(std::make_shared<v0::Log>(division))};
+};
+
+OutputVector translate_special_logit(const NodeContext& context) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/logit.cpp:41]: This is the same code as in `translate_logit`. No need to copy it, just call `translate_logit` in `op_table`

File: src/frontends/pytorch/src/op/logit.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    num_inputs_check(context, 1, 3);
+    auto x = context.get_input(0);
+    if (!context.input_is_none(1)) {
+        auto eps = context.get_input(1);
+        auto one = context.mark_node(v0::Constant::create(element::f32, Shape{}, {1}));
+        auto one_minus_eps = context.mark_node(std::make_shared<v1::Subtract>(one, eps));
+        eps = context.mark_node(std::make_shared<v1::ConvertLike>(eps, x));
+        x = context.mark_node(std::make_shared<v1::Maximum>(x, eps));
+        one_minus_eps = context.mark_node(std::make_shared<v1::ConvertLike>(one_minus_eps, x));
+        x = context.mark_node(std::make_shared<v1::Minimum>(x, one_minus_eps));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/logit.cpp:31]: Instead of `Maximum` and `Minimum`, just use `Clamp`

**Review by mvafin — CHANGES_REQUESTED** [review]
Thank you for the PR. Please add tests.

File: tests/layer_tests/pytorch_tests/test_logit.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import pytest
+import torch
+import numpy as np
+
+from pytorch_layer_test_class import PytorchLayerTest
+
+class aten_logit(torch.nn.Module):
+    def __init__(self, eps=None):
+        self.eps = eps
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_logit.py:12]:
```suggestion
        super().__init__()
        self.eps = eps
```

**Comment by mvafin** [comment]
The tests fail with accuracy, please check the formula

**Comment by code-dev05** [comment]
I tested it myself and what I can see is that -
- For CPU, FP32 is failing and FP16 is passing
- For GPU, both FP32 and FP16 are failing
I am unable to understand the reasoning behind this. It would be helpful if you could provide some insight.

**Comment by code-dev05** [comment]
I tried the following piece of code - 
`import torch
from openvino.frontend.pytorch.ts_decoder import TorchScriptPythonDecoder
import openvino as ov
import numpy as np

device = "cuda" if torch.cuda.is_available() else "cpu" 
print(device)
class Model(torch.nn.Module):
    def __init__(self, eps=None):
        super().__init__()
        self.eps = eps

    def forward(self, x):
        return torch.logit(x, self.eps)
    
model = Model(1e-6).to(device)
model.eval()
example = torch.rand((2, 3))

traced_model = torch.jit.trace(model, example)
print(traced_model.inlined_graph)

scripted_model = torch.jit.script(model)
print(scripted_model.inlined_graph)

decoder = TorchScriptPythonDecoder(model, example_input=example)
print(decoder.graph_element)

ov_model = ov.convert_model(model, example_input=example)
core = ov.Core()
compiled = core.compile_model(ov_model, "GPU" if device == 'cuda' else "CPU")

pt_res = model(example)
ov_res = compiled(example)
np.testing.assert_allclose(pt_res.detach().numpy(), ov_res[0], atol=1e-4, rtol=1e-4)`

Since I am using `torch.randn` example is of type f32
This code works correctly for cpu but not for gpu.

**Comment by mvafin** [comment]
> I tried the following piece of code -
> 
> ```
>   import torch
>   from openvino.frontend.pytorch.ts_decoder import TorchScriptPythonDecoder
>   import openvino as ov
>   import numpy as np
>   
>   device = "cuda" if torch.cuda.is_available() else "cpu" 
>   print(device)
>   class Model(torch.nn.Module):
>       def __init__(self, eps=None):
>           super().__init__()
>           self.eps = eps
>   
>       def forward(self, x):
>           return torch.logit(x, self.eps)
>       
>   model = Model(1e-6).to(device)
>   model.eval()
>   example = torch.rand((2, 3))
>   
>   traced_model = torch.jit.trace(model, example)
>   print(traced_model.inlined_graph)
>   
>   scripted_model = torch.jit.script(model)
>   print(scripted_model.inlined_graph)
>   
>   decoder = TorchScriptPythonDecoder(model, example_input=example)
>   print(decoder.graph_element)
>   
>   ov_model = ov.convert_model(model, example_input=example)
>   core = ov.Core()
>   compiled = core.compile_model(ov_model, "GPU" if device == 'cuda' else "CPU")
>   
>   pt_res = model(example)
>   ov_res = compiled(example)
>   np.testing.assert_allclose(pt_res.detach().numpy(), ov_res[0], atol=1e-4, rtol=1e-4)
> ```
> 
> Since I am using `torch.randn` example is of type f32 This code works correctly for cpu but not for gpu.

Please focus on CPU for now. The gpu might have accuracy issues as it does calculations in FP16 by default

**Comment by code-dev05** [comment]
Can you suggest what should i look into to solve the accuracy issue? 
Since the f32 is failing for cpu but f16 is passing.

@mvafin

**Comment by mvafin** [comment]
> Can you suggest what should i look into to solve the accuracy issue? Since the f32 is failing for cpu but f16 is passing.
> 
> @mvafin

fp16 only pass because it has less strict thresholds. Please make sure CPU FP32 tests pass

**Comment by code-dev05** [comment]
Hello @mvafin ,
Could you tell me the fix for this accuracy issue?
